teamscale-maven-plugin/src/main/java/com/teamscale/maven/upload/CoverageUploadMojo.java

```diff
@@ -0,0 +1,196 @@
+		collectReportOutputDirectory(session.getTopLevelProject(), "report-aggregate", "jacoco-aggregate", reportAggregateGoalOutputFiles);
+		getLog().debug(String.format("Found %d sub-modules", session.getTopLevelProject().getCollectedProjects().size()));
+		for (MavenProject subProject : session.getTopLevelProject().getCollectedProjects()) {
+			collectReportOutputDirectory(subProject, "report", "jacoco", reportGoalOutputFiles);
+			collectReportOutputDirectory(subProject, "report-integration", "jacoco-it", reportIntegrationGoalOutputFiles);
+			collectReportOutputDirectory(subProject, "report-aggregate", "jacoco-aggregate", reportAggregateGoalOutputFiles);
+		}
+	}
+
+	/**
+	 * Collect the file locations in which JaCoCo is configured to save the XML coverage reports
+	 * @param project The project
+	 * @param reportGoal The JaCoCo report goal
+	 * @param jacocoDirectory The name of the directory, matching the JaCoCo goal
+	 * @see <a href="https://www.eclemma.org/jacoco/trunk/doc/maven.html">Goals</a>
+	 */
+	private void collectReportOutputDirectory(MavenProject project, String reportGoal, String jacocoDirectory, List<Path> reportOutputFiles) throws MojoFailureException {
+		Path defaultOutputDirectory = Paths.get(project.getReporting().getOutputDirectory());
+		// If a Dom is null it means the execution goal uses default parameters which work correctly
+		Xpp3Dom reportConfigurationDom = getJacocoGoalExecutionConfiguration(project,reportGoal);
+		String errorMessage = "Skipping upload for %s as %s is not configured to produce XML reports for goal %s. See https://www.jacoco.org/jacoco/trunk/doc/report-mojo.html#formats";
+		if (!validateReportFormat(reportConfigurationDom)) {
+			throw new MojoFailureException(String.format(errorMessage, project.getName(), JACOCO_PLUGIN_NAME, jacocoDirectory));
+		}
+		Path resolvedOutputDir = getCustomOutputDirectory(reportConfigurationDom).orElse(defaultOutputDirectory.resolve(jacocoDirectory).resolve("jacoco.xml"));
+		getLog().debug(String.format("Adding possible report location: %s", resolvedOutputDir));
+		reportOutputFiles.add(resolvedOutputDir);
+	}
+
+	private void uploadCoverageReports() throws IOException {
+		uploadCoverage(reportGoalOutputFiles, unitTestPartition);
+		uploadCoverage(reportIntegrationGoalOutputFiles, integrationTestPartition);
+		uploadCoverage(reportAggregateGoalOutputFiles, aggregatedTestPartition);
+	}
+
+	private void uploadCoverage(List<Path> reportOutputFiles, String partition) throws IOException {
+		List<File> reports = new ArrayList<>();
+		getLog().debug(String.format("Scanning through %d locations for %s...", reportOutputFiles.size(), partition));
+		for (Path reportPath : reportOutputFiles) {
+			File report = reportPath.toFile();
+			if (!report.exists()) {
+				getLog().debug(String.format("Cannot find %s, skipping...", report.getAbsolutePath()));
```

Comment:
```suggestion
				getLog().warn(String.format("Cannot find %s, skipping...", report.getAbsolutePath()));
```
I think this should be a warning aswell.

Reply:
Ah okay that makes sense, thanks for the explanation!